src/core/server/services/users/user.updateRole.spec.ts

```diff
@@ -0,0 +1,209 @@
+  moderationScopes: {
+    siteIDs: [site1.id],
+  },
+});
+const site1Member = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.MEMBER,
+  membershipScopes: {
+    siteIDs: [site1.id],
+  },
+});
+const site1Commenter = createUserFixture({
+  tenantID,
+  role: GQLUSER_ROLE.COMMENTER,
+});
+const users = [
+  admin,
+  orgMod,
+  site1Mod,
+  site1Staff,
+  site1Member,
+  site1Commenter,
+];
+
+const inputStr = (
+  viewerRole: GQLUSER_ROLE,
+  userFromRole: GQLUSER_ROLE,
+  userToRole: GQLUSER_ROLE
+) =>
+  `viewer role = ${viewerRole}, user from role = ${userFromRole},  userToRole = ${userToRole}`;
+
+describe("updateRole", () => {
+  let mongo: MongoContext;
+
+  /* eslint-disable-next-line */
+  require("coral-server/models/user").retrieveUser.mockImplementation(
+    async (_mongo: any, _tenantID: any, userID: string) =>
+      users.find(({ id }) => id === userID)
+  );
+
+  /* eslint-disable-next-line */
+  require("coral-server/models/user").updateUserRole.mockImplementation(
+    async (
+      _mongo: any,
+      _tenantID: any,
+      userID: string,
+      role: GQLUSER_ROLE
+    ) => ({
+      ...users.find(({ id }) => id === userID),
+      role,
+    })
+  );
+
+  let uut: (
+    mongo: MongoContext,
+    tenant: Tenant,
+    viewer: User,
+    user: User,
+    role: GQLUSER_ROLE
+  ) => ReturnType<typeof updateRole>;
+
+  const exhaustiveInputs = new Map<string, boolean>();
+  beforeAll(() => {
+    ({ ctx: mongo } = createMockMongoContex());
+
+    // Wrapping our function to track what inputs have been tested
+    uut = (mongoCtx, tenant, viewer, user, role) => {
+      const inputHash = inputStr(viewer.role, user.role, role);
+      exhaustiveInputs.set(inputHash, true);
+
+      return updateRole(mongo, tenant, viewer, user.id, role);
+    };
+
+    for (const viewer of users) {
+      for (const user of users) {
+        for (const newRole of ROLES) {
+          const inputHash = inputStr(viewer.role, user.role, newRole);
+          exhaustiveInputs.set(inputHash, false);
+        }
+      }
+    }
+  });
+
+  it("allows admins to assign any user any role", async () => {
+    const nonAdmins = users.filter(({ role }) => role !== GQLUSER_ROLE.ADMIN);
+    for (const user of nonAdmins) {
+      for (const role of ROLES) {
+        const res = await uut(mongo, mockTenant, admin, user, role);
+
+        expect(res.role).toEqual(role);
+      }
+    }
+  });
+
+  it("TODO: is this ok?! allows organization moderators to allocate site mods", async () => {
+    const nonMods = [site1Staff, site1Member, site1Commenter];
+    for (const user of nonMods) {
+      const res = await uut(
+        mongo,
+        mockTenant,
+        orgMod,
+        user,
+        GQLUSER_ROLE.MODERATOR
+      );
+
+      expect(res.role).toEqual(GQLUSER_ROLE.MODERATOR);
+    }
+  });
+
+  it("does not allow site mods, staff, members, or commenters to change anyones role", async () => {
```

Comment:
Site mods are able to make other users site mods for their sites and members though, right?

Reply:
i don’t believe so, i think only organization moderators can do that. otherwise community mods for SBN could elevate their friends to moderators. i’ll confirm in standup tomorrow though

Reply:
I wanted to follow up on this. If you go into the `Community` tab in the admin, site moderators currently can change users' roles to site moderator or member for their site(s), so I think this test should be updated.

Reply:
✅